### ISSUE TITLE
Stuck on booting worker

### ISSUE THREAD

**Issue by Levelleor** [voxpupuli/puppetboard#656]
Following the Readme I'm putting the following on the puppetdb machine. It is separate from the master. 
```
root@puppet-db:~# docker run -it -p 9080:80 -v /etc/puppetlabs/puppet/ssl:/etc/puppetlabs/puppet/ssl \
> -e PUPPETDB_HOST=localhost \
> -e PUPPETDB_PORT=8081 \
> -e PUPPETDB_SSL_VERIFY=/etc/puppetlabs/puppetdb/ssl/ca.pem \
> -e PUPPETDB_KEY=/etc/puppetlabs/puppetdb/ssl/private.pem \
> -e PUPPETDB_CERT=/etc/puppetlabs/puppetdb/ssl/public.pem \
> ghcr.io/voxpupuli/puppetboard
[2022-01-13 10:52:13 +0000] [1] [INFO] Starting gunicorn 20.0.0
[2022-01-13 10:52:13 +0000] [1] [INFO] Listening at: http://0.0.0.0:80 (1)
[2022-01-13 10:52:13 +0000] [1] [INFO] Using worker: sync
[2022-01-13 10:52:13 +0000] [11] [INFO] Booting worker with pid: 11
```

It gets stuck on booting here and never goes past it. I see no error messages. From what I understand this Docker image contains PuppetDB and PostgreSQL servers setup and routes traffic from port 80 to Docker container. Is there additional setup required for this to work? What kind of setup?

**Comment by gdubicki** [comment]
Hi @Levelleor ! Thanks for reporting the issue and sorry for the trouble.

I am not sure how it got to this as I have been only moving it around and reformating it, but the command we provided in the README in my opinion could never work. :(

**tl;dr:** try running this command instead, on your PuppetDB server:
```
docker run -it -e PUPPETDB_HOST=localhost -e PUPPETDB_PORT=8080 --net=host ghcr.io/voxpupuli/puppetboard
```
...and then in another terminal on the PuppetDB server try curling http://127.0.0.1 .

(Why?

Firstly the volume path in `-v` in the command that it's the README doesn't match the path of the certs passed later in the env variables (`-e`).

Secondly without using host networking (`--net=host`) you cannot reference to the host (PuppetDB server) using `localhost` inside the container as that means the container there...

Lastly using SSL will not really work because your PuppetDB certname will probably not be `localhost` and you will get sth like:
```
ERROR:puppetboard.utils:HTTPSConnectionPool(host='localhost', port=8081): Max retries exceeded with url: /pdb/query/v4/environments (Caused by SSLError(SSLCertVerificationError("hostname 'localhost' doesn't match '<your puppetdb FQDN>'")))
```
)

Let's see if it helps and we'll move forward from there. :)

**Comment by Levelleor** [comment]
Hi @gdubicki, first of all, thank you for updating! Ok, so I do get this error. I didn't really find any documentation as to what PUPPETDB_HOST is supposed to be. Could you shine some light on this? 

> Lastly using SSL will not really work because your PuppetDB certname will probably not be localhost and you will get sth like:
>
> ERROR:puppetboard.utils:HTTPSConnectionPool(host='localhost', port=8081): Max retries exceeded with url: /pdb/query/v4/environments (Caused by SSLError(SSLCertVerificationError("hostname 'localhost' doesn't match '<your puppetdb FQDN>'")))

**Comment by gdubicki** [comment]
Have you tried using the command I suggested above, @Levelleor ? Does it work for you?

**Comment by Levelleor** [comment]
@gdubicki yes, I get the issue above from the command you've provided.

**Comment by gdubicki** [comment]
Try this then:
```
docker run -it -e PUPPETDB_HOST=localhost -e PUPPETDB_PORT=8081 -e PUPPETDB_KEY=/etc/puppetlabs/puppetdb/ssl/private.pem -e PUPPETDB_CERT=/etc/puppetlabs/puppetdb/ssl/public.pem -e PUPPETDB_SSL_VERIFY=False --net=host -v /etc/puppetlabs/puppetdb/ssl:/etc/puppetlabs/puppetdb/ssl ghcr.io/voxpupuli/puppetboard
```

**Comment by Levelleor** [comment]
@gdubicki Thank you, it did something. It doesn't look like it's crashing anymore, now it simply says "Service Unavailable". Here're logs from a curl: 

> /usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 'localhost'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
>   warnings.warn(
> WARNING:charset_normalizer:override steps (5) and chunk_size (512) as content does not fit (44 byte(s) given) parameters.
> INFO:charset_normalizer:ascii passed initial chaos probing. Mean measured chaos is 0.000000 %
> INFO:charset_normalizer:ascii should target any language(s) of ['Latin Based']
> INFO:charset_normalizer:We detected language [('English', 1.0), ('Dutch', 1.0), ('Polish', 0.9412)] using ascii
> INFO:charset_normalizer:ascii is most likely the one. Stopping the process.
> ERROR:pypuppetdb.api.base:PuppetDB is currently down. Try again later. localhost:8081 over HTTPS.
> ERROR:puppetboard.utils:503 Server Error: Service Unavailable for url: https://localhost:8081/pdb/query/v4/environments
> 127.0.0.1 - - [18/Jan/2022:17:14:20 +0000] "GET / HTTP/1.1" 503 262 "-" "curl/7.68.0"

**Comment by gdubicki** [comment]
Sorry but at this point you need to check what is the proper port and protocol (HTTP/HTTPS) for your PuppetDB API and update one of the commands I gave you accordingly.

The first command I gave you is for plain HTTP and the second for HTTPS.

**Comment by Levelleor** [comment]
@gdubicki What do you mean the proper port? Where does that come from? Do I need to setup PuppetDB manually as well for this to work?

**Comment by gdubicki** [comment]
You need to already have PuppetDB itself set up and working, if that's what you meant. That is a requirement for using Puppetboard as its only a UI for the former.

**Comment by Levelleor** [comment]
Oh.. I see. I though this would install PuppetDB and PostgreSQL within a container without any additional setup required. Well, I guess back to troubleshooting PuppetDB and this isn't relevant anymore.

**Comment by Levelleor** [comment]
Ok, I don't know what's the issue now but I got PuppetDB up yet now it says this: 

> <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
> <title>404 Not Found</title>
> <h1>Not Found</h1>
> <p>The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.</p>

The PuppetDB is now working and I am lost again to finding the root cause.

**Comment by gdubicki** [comment]
Is your PuppetDB working and your nodes are publishing reports to it? Do you have nodes in the default environment "production" ? I am asking because perhaps you are hit by #515.
